### ISSUE TITLE
Using autocast leads to the training speed becomes slower

### ISSUE THREAD

**Issue by HelloWorldLTY** [pyg-team/pytorch_geometric#6576]
### 🐛 Describe the bug

Hi, I tried to use autocast in the training process of GNN, although I find the memory I used became lower, the training speed becomes longer than the previous design, like 2 times. Are there any reasons for causing this problem? Are there any people who face the same problem? Thanks a lot.

### Environment

* PyG version: 1.7
* PyTorch version: 1.11.0
* OS: linux
* Python version: 3.8
* CUDA/cuDNN version: 11.7
* How you installed PyTorch and PyG (`conda`, `pip`, source):
* Any other relevant information (*e.g.*, version of `torch-scatter`):

**Comment by rusty1s** [comment]
I don't have much experience with `torch.amp`, so I cannot answer this right away. Do you have a small example to reproduce this? Happy to look into.

**Comment by LukeLIN-web** [comment]
Maybe model converage needs more time.

**Comment by HelloWorldLTY** [comment]
Hi, I can attach pseudo code here:
I combine autocast and scaler with current pyg codes. 
lambda_infonce = 0.001
scaler = GradScaler(growth_interval=100)  # helps to increase _scale back to a larger number more often
min_scale = 128
for epoch in range(3000):
     
    for i in range(0,len(graph_list)):
        
        optimizer_enc_is.zero_grad(set_to_none=True)
        optimizer_enc_com.zero_grad(set_to_none=True)
        optimizer_dec2.zero_grad(set_to_none=True)
        
        graph = graph_list[i]
        
        x = graph.x.to(device)
        train_pos_edge_index = graph.edge_index.long().to(device)
        edge_adj = torch.FloatTensor(cor_list[i].values).to(device)
        
        with autocast():
            x = gene_encoder_is(x, train_pos_edge_index, graph.show_index)
            z = gene_encoder_com(x, train_pos_edge_index, graph.show_index)
            adj = torch.matmul(z, z.t())
            edge_reconstruct = gene_decoder(adj, graph.show_index).to(torch.float32)
            loss = loss_f(edge_reconstruct.flatten(), edge_adj.flatten())
       
        del graph
        scaler.scale(loss).backward()
        del loss
        
        scaler.step(optimizer_enc_is)
        scaler.step(optimizer_enc_com)
        scaler.step(optimizer_dec2)
        scaler.update()
        if scaler._scale < min_scale:
            scaler._scale = torch.tensor(min_scale).to(scaler._scale)
